pandas/tests/scalar/timedelta/test_timedelta.py

```diff
@@ -348,9 +348,20 @@ def test_unit_parser(self, units, np_unit, wrapper):
 
             result = to_timedelta('2{}'.format(unit))
             assert result == expected
-            result = Timedelta('2{}'.format(unit))
+            result = Timedelta('2{}'.format(units))
             assert result == expected
 
+    @pytest.mark.parametrize('units', ['Y', 'y', 'M'])
+    def test_unit_m_y_deprecated(self, units):
+        with pytest.warns(FutureWarning) as record1:
+            to_timedelta(10, units)
+        assert len(record1) == 1
+        assert str(record1[0].message) == "M and Y units are deprecated."
+        with pytest.warns(FutureWarning) as record2:
+            Timedelta(10, units)
```

Comment:
You can also parameterize over `Timedelta`, `to_timedelta`. Could you also test `TimedeltaIndex` as well?